install.sh

```diff
@@ -1,20 +1,57 @@
 #!/bin/bash
 
-if ! which make > /dev/null; then
-   echo -e "make not found! Install it along with git? (y/n) \c"
-   read
-   if "$REPLY" = "y"; then
-      sudo apt-get install -y git make
-   fi
+function error() {
+   echo -e "\e[1;31m$1\e[0m"
+   exit 1
+}
+
+if ! command -v make > /dev/null; then
+   while true; do
+      echo -e "make not found! Install it? (y/n) \c"
+      read -r answer
+      if [[ "$answer" =~ [yY] ]]; then
+         sudo apt install -y make || error "Failed to install make!"
+         break
+      else
+         echo -e "\e[1;31mInvalid answer \"$REPLY\"!\e[0m"
+      fi
+   done
+fi
+
+if ! command -v git > /dev/null; then
+   while true; do
+      echo -e "git not found! Install it? (y/n) \c"
+      read -r answer
+      if [[ "$answer" =~ [yY] ]]; then
+         sudo apt install -y git || error "Failed to install git!"
+         break
+      else
+         echo -e "\e[1;31mInvalid answer \"$REPLY\"!\e[0m"
+      fi
+   done
 fi
 
-git clone https://github.com/CleanMachine1/DollarSkip
-cd DollarSkip
-make 
-sudo make install
+if [[ ! -d DollarSkip ]]; then
+   git clone https://github.com/CleanMachine1/DollarSkip.git || error "Failed to clone DollarSkip repository!"
+else
+   while true; do
+      echo -e "\e[1m\"DollarSkip\" folder already exists, would you like to update it (git pull)?\e[0m"
+      read -r answer
+      if [[ "$answer" =~ [yY] ]]; then
+         cd DollarSkip || error "Failed to enter DollarSkip folder!"
+         git pull || error "Failed to run \"git pull\"!"
+         break
+      else
+         break
+      fi
+   done
+fi
```

Comment:
```suggestion
fi
if [[ $(pwd | grep DollarSkip -c) = 0 ]]; then
   cd DollarSkip || error "test"
fi
```

Reply:
This fixes the issues

Reply:
I fixed them in a way I feel more comfortable with. feel free to change to this if you think its better.